From 0f5769ea6856ccd500018439c1bfb6a61d29c584 Mon Sep 17 00:00:00 2001
From: Ethan Chen <intervigil@gmail.com>
Date: Sun, 20 May 2018 14:37:47 -0700
Subject: [PATCH 11/12] build: Clean up makefile inclusions

* Include MTK/QCOM rules directly from the build/core/config.mk hook.
* Place main.mk hook and main_version.mk into build/core with the rest
  of the make system hooks.

Change-Id: Iad2f808f3e3eefbd2be2ca5facb907ff31babbb7
---
 build/core/config.mk                   | 6 ++++++
 build/core/main.mk                     | 2 ++
 {config => build/core}/main_version.mk | 0
 config/main.mk                         | 2 --
 4 files changed, 8 insertions(+), 2 deletions(-)
 create mode 100644 build/core/main.mk
 rename {config => build/core}/main_version.mk (100%)
 delete mode 100644 config/main.mk

diff --git a/build/core/config.mk b/build/core/config.mk
index 5f46cab0..71a73d0e 100644
--- a/build/core/config.mk
+++ b/build/core/config.mk
@@ -21,3 +21,9 @@ FRAMEWORK_LINEAGE_PLATFORM_REMOVED_API_FILE := $(TOPDIR)lineage-sdk/api/lineage_
 FRAMEWORK_LINEAGE_API_NEEDS_UPDATE_TEXT := $(TOPDIR)vendor/lineage/build/core/apicheck_msg_current.txt
 
 BUILD_RRO_SYSTEM_PACKAGE := $(TOPDIR)vendor/lineage/build/core/system_rro.mk
+
+# Rules for MTK targets
+include $(TOPDIR)vendor/lineage/build/core/mtk_target.mk
+
+# Rules for QCOM targets
+include $(TOPDIR)vendor/lineage/build/core/qcom_target.mk
diff --git a/build/core/main.mk b/build/core/main.mk
new file mode 100644
index 00000000..0793b8a7
--- /dev/null
+++ b/build/core/main.mk
@@ -0,0 +1,2 @@
+# Include LineageOS versions
+include $(TOPDIR)vendor/lineage/build/core/main_version.mk
diff --git a/config/main_version.mk b/build/core/main_version.mk
similarity index 100%
rename from config/main_version.mk
rename to build/core/main_version.mk
diff --git a/config/main.mk b/config/main.mk
deleted file mode 100644
index 038a8469..00000000
--- a/config/main.mk
+++ /dev/null
@@ -1,2 +0,0 @@
-# Include LineageOS versions
-include vendor/lineage/config/main_version.mk
-- 
2.17.0

